Fix 5 - Integrate frequency attribute in database

jean553 opened the pull request.

jean553 committed:
re #fix_5 add frequency to each CSV file entry

jean553 committed:
re #fix_5 edit lib.rs to be used with DBEntry and not String anymore

re #fix_5 the HashMap of the database is now an array of DBEntity
objects and not an array of String

re #fix_5 add dbentity in mod.rs

re #fix_5 add dbentry structure who contains the sinograms and frequency
of one entry in database

re #fix_5 delete useless code

re #fix_5 update the dbentry structure : add a new() method

jean553 committed:
re #fix_5 get the frequency of one sinogram from the database

re #fix_5 fix entry copy problem

allan-simon commented on src/pinyin/db.rs:
```diff
@@ -2,21 +2,23 @@ extern crate serialize;
 use self::serialize::json;
-use std::collections::HashMap;
+use std::collections::{HashMap, hashmap};
+use std::collections::hashmap::{Occupied, Vacant};
```
this line is not used , can be removed

allan-simon commented on src/pinyin/db.rs:
```diff
@@ -2,21 +2,23 @@ extern crate serialize;
 use std::hash::sip::SipHasher;
 use pinyin::myfile::open_read_only;
 
+use pinyin::dbentry::DBEntry;
```
it seems people are to write this kind of things  as  DbEntry

allan-simon commented on src/pinyin/db.rs:
```diff
@@ -30,12 +32,17 @@ pub fn create_db_from_csv(fname: &str) -> PinyinDB {
         let mut iter = line.as_slice().split(',');
         let sinogram =  iter.next().unwrap();
```
put the .to_string() here , as you will always use it as a string , which will make clearer the code after

allan-simon commented on src/pinyin/db.rs:
```diff
@@ -30,12 +32,17 @@ pub fn create_db_from_csv(fname: &str) -> PinyinDB {
         let mut iter = line.as_slice().split(',');
         let sinogram =  iter.next().unwrap();
         let pinyin = iter.next().unwrap();
```
same for pinyin here

allan-simon commented on src/pinyin/db.rs:
```diff
@@ -69,21 +76,27 @@ pub fn create_db(fname: &str) -> PinyinDB {
-                |_key, value| value.push(sinogram.clone())
+
+            let full_entry = DBEntry::new(
+                sinogram.to_string(),
```
same here for sinogram, move ithe "to_string"  before , so that you don't need to do it twice

jean553 committed:
re #fix_5 change structure name

re #fix_5 update the dbentry structure name and refractor code

re #fix_5 delete unused module
